pkg/manifests/manifests.go

```diff
@@ -1416,8 +1436,199 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
 	return p, nil
 }
 
-func (f *Factory) AdditionalAlertManagerConfigsSecret() (*v1.Secret, error) {
-	alertmanagerConfigs := f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs
+func (f *Factory) PrometheusK8sAdditionalAlertManagerConfigsSecret() (*v1.Secret, error) {
+	amConfigs := f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.AlertmanagerConfigs
+
+	config, err := f.additionalAlertManagerConfigs(amConfigs, alertmanagerHttpConfigFormatPrometheus)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PrometheusK8sAdditionalAlertmanagerConfigSecretName,
+			Namespace: f.namespace,
+		},
+		Data: map[string][]byte{
+			AdditionalAlertmanagerConfigSecretKey: config,
+		},
+	}, nil
+}
+
+func (f *Factory) PrometheusUserWorkloadAdditionalAlertManagerConfigsSecret() (*v1.Secret, error) {
+	amConfigs := f.config.GetPrometheusUWAdditionalAlertmanagerConfigs()
+	config, err := f.additionalAlertManagerConfigs(amConfigs, alertmanagerHttpConfigFormatPrometheus)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PrometheusUWAdditionalAlertmanagerConfigSecretName,
+			Namespace: f.namespaceUserWorkload,
+		},
+		Data: map[string][]byte{
+			AdditionalAlertmanagerConfigSecretKey: config,
+		},
+	}, nil
+}
+
+func (f *Factory) prometheusAlertmanagerConfigHttpSection(alertmanagerConfig AdditionalAlertmanagerConfig) ([]yaml2.MapItem, error) {
```

Comment:
I had quite a hard time understanding the difference between `prometheusAlertmanagerConfigHttpSection` and `thanosAlertmanagerConfigHttpSection` at first. To make it clearer, what do you think about refactoring the part where we get the bearerToken and TLSConfig and have custom marshal functions to translate into the different config based on the upstream types?

Reply:
Ack, will put some effort into refactoring the code a bit.

Reply:
I extracted the yaml rendering part here: https://github.com/openshift/cluster-monitoring-operator/pull/1271/files as part of a new commit.

Please take a look again and let me know what you think.

Reply:
awesome! it is way easier to understand now :100: